q2_fmt/_peds.py

```diff
@@ -8,30 +8,191 @@
+                peds = num_sum[count] / donor_sum[count]
+
+            peds_df.loc[len(peds_df)] = [sample, peds, num_sum[count],
+                                         donor_sum[count],
+                                         sample_row[reference_column],
+                                         sample_row[subject_column],
+                                         sample_row[time_column]]
+        peds_df['id'].attrs.update({
+            'title': reference_series.index.name,
+            'description': 'Sample IDs'
+        })
+        peds_df['measure'].attrs.update({
+            'title': "PEDS",
+            'description': 'Proportional Engraftment of Donor Strains '
+        })
+        peds_df['group'].attrs.update({
+            'title': time_column,
+            'description': 'Time'
+        })
+        peds_df["subject"].attrs.update({
+            'title': subject_column,
+            'description': 'ID to link samples across time'
+        })
+        peds_df["transfered_donor_features"].attrs.update({
+            'title': "Transfered Donor Features",
+            'description': '...'
+        })
+        peds_df['total_donor_features'].attrs.update({
+            'title': "Total Donor Features",
+            'description': '...'
+        })
+        peds_df['donor'].attrs.update({
+            'title': reference_column,
+            'description': 'Donor'
+        })
+
+    elif peds_type == "Feature":
+        num_sum = np.sum(maskedrecip, axis=0)
+        donor_sum = np.sum(donormask, axis=0)
+        for count, feature in enumerate(recipdf.columns):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                peds = num_sum[count] / donor_sum[count]
+            peds_df.loc[len(peds_df)] = [feature, peds, num_sum[count],
+                                         donor_sum[count], peds_time, feature]
+            peds_df = peds_df.dropna()
+        peds_df['id'].attrs.update({
+            'title': "FeatureID",
+            'description': ''
+        })
+        peds_df['measure'].attrs.update({
+            'title': "PEDS",
+            'description': 'Proportional Engraftment of Donor Strains '
+        })
+        peds_df['group'].attrs.update({
+            'title': time_column,
+            'description': 'Time'
+        })
+        peds_df['subject'].attrs.update({
+            'title': "FeatureID",
+            'description': ''
+        })
+    else:
+        raise KeyError('There was an error finding which PEDS methods to use')
+    return peds_df
+
+
+# Filtering methods
+def _check_for_time_column(metadata, time_column):
     try:
         num_timepoints = metadata[time_column].dropna().unique().size
     except Exception as e:
         raise KeyError('There was an error finding %s in the metadata'
                        % time_column) from e
+    return num_timepoints
+
+
+def _check_time_column_numeric(metadata, time_column):
```

Comment:
I'm interested in what @ebolyen thinks about this - but I'm wondering if you can just combine the two filtering methods for the time column - checking for existence, and then the type (ensuring numeric) before returning it from the `_check_for_time_column` helper method.

Reply:
I am open to merging them. I just kinda like that it was only checking one specific thing. Made more sense in my head